src/views/connectionUI.ts

```diff
@@ -195,21 +195,49 @@ export class ConnectionUI {
                 };
             });
 
+            // Add an option to disconnect from the current server
+            const disconnectItem: vscode.QuickPickItem = {
+                label: LocalizedConstants.disconnectOptionLabel,
+                description: LocalizedConstants.disconnectOptionDescription
+            };
+            pickListItems.push(disconnectItem);
+
             const pickListOptions: vscode.QuickPickOptions = {
                 placeHolder: LocalizedConstants.msgChooseDatabasePlaceholder
             };
 
             // show database picklist, and modify the current connection to switch the active database
-            self.vscodeWrapper.showQuickPick<Interfaces.IConnectionCredentialsQuickPickItem>(pickListItems, pickListOptions).then( selection => {
+            self.vscodeWrapper.showQuickPick<vscode.QuickPickItem>(pickListItems, pickListOptions).then( selection => {
+                if (selection === disconnectItem) {
+                    self.handleDisconnectChoice().then(() => resolve(undefined), err => reject(err));
+                }
                 if (typeof selection !== 'undefined') {
```

Comment:
Shouldn't this be an else if? While resolve will eventually return from the promise it doesn't actually block the subsequent code from being run I think?